python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -1461,7 +1461,9 @@ def create_run(
         check.opt_set_param(asset_selection, "asset_selection", of_type=AssetKey)
         check.opt_set_param(asset_check_selection, "asset_check_selection", of_type=AssetCheckKey)
 
-        if asset_selection is not None or asset_check_selection is not None:
+        # asset_selection gets coerced from [] to None, but asset_check_selection doesn't. We
+        # allow asset_check_selection to be [] when op_selection is set.
```

Comment:
I'm still struggling to follow this. Where does the coercing being referenced here happen? Can we add some whys in addition to some whats here / be more verbose?